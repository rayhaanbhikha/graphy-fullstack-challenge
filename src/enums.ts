
export enum ApplicationState {
  EDIT_MODE,
  DEFAULT_MODE
}

export enum AnnotationStates {
  OPEN,
  CLOSED,
  EDITING,
  DELETING,
  DRAGGING
}