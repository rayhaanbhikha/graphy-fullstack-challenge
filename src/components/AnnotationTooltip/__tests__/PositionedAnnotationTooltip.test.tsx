import React from 'react';
import { render } from '@testing-library/react'
import { PositionedAnnotationTooltip } from '../PositionedAnnotationTooltip';
import 'jest-styled-components';
import { AnnotationStates } from '../../../enums';

describe('PositionedAnnotationTooltip', () => {

  it('should have correct default styles', () => {
    const { asFragment } = render(<PositionedAnnotationTooltip annotationState={AnnotationStates.CLOSED} />);
    expect(asFragment()).toMatchSnapshot();
  })

  it('should have correct styles when in edit state', () => {
    const { asFragment } = render(<PositionedAnnotationTooltip annotationState={AnnotationStates.EDITING} />);
    expect(asFragment()).toMatchSnapshot();
  })

  it('should have correct styles when in open state', () => {
    const { asFragment } = render(<PositionedAnnotationTooltip annotationState={AnnotationStates.OPEN} />);
    expect(asFragment()).toMatchSnapshot();
  })
});