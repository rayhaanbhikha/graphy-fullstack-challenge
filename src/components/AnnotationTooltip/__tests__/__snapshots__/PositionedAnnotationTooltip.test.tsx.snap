// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionedAnnotationTooltip should have correct default styles 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  left: 19px;
  top: 0;
  padding-left: 5px;
  visibility: hidden;
  opacity: 0;
  -webkit-transition: visibility 0.5s,opacity 0.5s ease;
  transition: visibility 0.5s,opacity 0.5s ease;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`PositionedAnnotationTooltip should have correct styles when in edit state 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  left: 19px;
  top: 0;
  padding-left: 5px;
  visibility: initial;
  opacity: 100;
  -webkit-transition: visibility 0.5s,opacity 0.5s ease;
  transition: visibility 0.5s,opacity 0.5s ease;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`PositionedAnnotationTooltip should have correct styles when in open state 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  left: 19px;
  top: 0;
  padding-left: 5px;
  visibility: initial;
  opacity: 100;
  -webkit-transition: visibility 0.5s,opacity 0.5s ease;
  transition: visibility 0.5s,opacity 0.5s ease;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
